src/Riok.Mapperly/Descriptors/Mappings/NewTupleExpressionMapping.cs

```diff
@@ -0,0 +1,30 @@
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Riok.Mapperly.Descriptors.Mappings.MemberMappings;
+
+namespace Riok.Mapperly.Descriptors.Mappings;
+
+/// <summary>
+/// An object mapping creating the target instance via a () expression,
+/// mapping properties via ctor, object initializer but not by assigning.
+/// <seealso cref="NewInstanceObjectMemberMethodMapping"/>
+/// </summary>
+public class NewTupleExpressionMapping : TypeMapping, INewTupleExpressionMapping
+{
+    private readonly HashSet<TupleConstructorParameterMapping> _constructorPropertyMappings = new();
+
+    public NewTupleExpressionMapping(ITypeSymbol sourceType, ITypeSymbol targetType)
+        : base(sourceType, targetType) { }
+
+    public void AddConstructorParameterMapping(TupleConstructorParameterMapping mapping) => _constructorPropertyMappings.Add(mapping);
+
+    public override ExpressionSyntax Build(TypeMappingBuildContext ctx)
+    {
+        // (Name:.. ,..);
+        var ctorArgs = _constructorPropertyMappings.Select(x => x.BuildArgument(ctx)).ToArray();
+        var objectCreationExpression = SyntaxFactory.TupleExpression(SyntaxFactory.SeparatedList(ctorArgs));
```

Comment:
This does not work in expressions, does it?

Reply:
I'm not sure what you mean, could you elaborate?

Reply:
Mapperly supports Mappings in expressions trees for Queryable projection mappings (see [docs](https://mapperly.riok.app/docs/configuration/queryable-projections)). AFAIK tuple expressions (eg. `(A: 10, B: 20)`) are not valid in expression trees (eg. this does not work: `Expression<Func<(int, int)>> x = () => (10, 20);`.
The above mapping is constructed even if the mapping takes place in an expression tree. I think this should be differentiated: in expression trees `new ValueTuple<int, int>(10, 20)` should be used, outside `(A: 10, B: 20)` should be used for increased readability.